consensus/src/liveness/proposal_generator.rs

```diff
@@ -195,12 +194,26 @@ impl PipelineBackpressureConfig {
                     }
                 })
                 .sorted()
-                // .sorted_by_key(|key| key.unwrap_or(u64::MAX))
                 .collect::<Vec<_>>();
-            info!("Estimated block back-offs block sizes: {:?}", sizes);
             if sizes.len() >= config.min_blocks_to_activate {
-                let index = (config.percentile * sizes.len() as f64) as usize;
-                sizes.get(index).copied()
+                let calibrated_block_size = *sizes
+                    .get(((config.percentile * sizes.len() as f64) as usize).min(sizes.len() - 1))
+                    .expect("guaranteed to be within vector size");
+                PROPOSER_ESTIMATED_CALIBRATED_BLOCK_TXNS.observe(calibrated_block_size as f64);
+                // Check if calibrated block size is reduction in size, to turn on backpressure.
+                if max_block_txns > calibrated_block_size {
```

Comment:
doesn't make a logical difference, but only "activating" backpressure if it is below the default block size - so counters for activating only show when it actually reduced the size